app/script/conversation/ConversationMapper.js

```diff
@@ -157,71 +157,89 @@ z.conversation.ConversationMapper = class ConversationMapper {
+        conversationEntity.set_timestamp(last_event_timestamp, z.conversation.TIMESTAMP_TYPE.LAST_EVENT);
       }
 
       if (last_read_timestamp) {
-        conversation_et.set_timestamp(last_read_timestamp, z.conversation.TIMESTAMP_TYPE.LAST_READ);
+        conversationEntity.set_timestamp(last_read_timestamp, z.conversation.TIMESTAMP_TYPE.LAST_READ);
       }
 
       if (last_server_timestamp) {
-        conversation_et.set_timestamp(last_server_timestamp, z.conversation.TIMESTAMP_TYPE.LAST_SERVER);
+        conversationEntity.set_timestamp(last_server_timestamp, z.conversation.TIMESTAMP_TYPE.LAST_SERVER);
       }
 
       if (muted_timestamp) {
-        conversation_et.set_timestamp(muted_timestamp, z.conversation.TIMESTAMP_TYPE.MUTED);
-        conversation_et.muted_state(self_state.muted_state);
+        conversationEntity.set_timestamp(muted_timestamp, z.conversation.TIMESTAMP_TYPE.MUTED);
+        conversationEntity = this.setMutedState(conversationEntity, selfState.muted_state);
       }
 
       if (status !== undefined) {
-        conversation_et.status(status);
+        conversationEntity.status(status);
       }
 
       if (verification_state !== undefined) {
-        conversation_et.verification_state(verification_state);
+        conversationEntity.verification_state(verification_state);
       }
 
       // Backend states
-      const {otr_archived, otr_muted} = self_state;
+      const {otr_archived, otr_muted} = selfState;
 
       if (otr_archived !== undefined) {
-        const otr_archived_timestamp = new Date(self_state.otr_archived_ref).getTime();
-        conversation_et.set_timestamp(otr_archived_timestamp, z.conversation.TIMESTAMP_TYPE.ARCHIVED);
-        conversation_et.archived_state(otr_archived);
+        const otrArchivedTimestamp = new Date(selfState.otr_archived_ref).getTime();
+        conversationEntity.set_timestamp(otrArchivedTimestamp, z.conversation.TIMESTAMP_TYPE.ARCHIVED);
+        conversationEntity.archived_state(otr_archived);
       }
 
       if (otr_muted !== undefined) {
-        const otr_muted_timestamp = new Date(self_state.otr_muted_ref).getTime();
-        conversation_et.set_timestamp(otr_muted_timestamp, z.conversation.TIMESTAMP_TYPE.MUTED);
-        conversation_et.muted_state(otr_muted);
+        const otrMutedTimestamp = new Date(selfState.otr_muted_ref).getTime();
+        conversationEntity.set_timestamp(otrMutedTimestamp, z.conversation.TIMESTAMP_TYPE.MUTED);
+        conversationEntity = this.setMutedState(conversationEntity, otr_muted, selfState.otr_muted_status);
       }
 
       if (disablePersistence) {
-        conversation_et.setStateChangePersistence(true);
+        conversationEntity.setStateChangePersistence(true);
       }
 
-      return conversation_et;
+      return conversationEntity;
+    }
+  }
+
+  setMutedState(conversationEntity, mutedStatus, deprecatedMutedState = false) {
```

Comment:
I think it would be safer to do the retro-compatibility adaptation at read time not write time (if I understand the code correctly)